pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompaction/UpsertCompactionTaskExecutor.java

```diff
@@ -117,8 +119,8 @@ protected SegmentConversionResult convert(PinotTaskConfig pinotTaskConfig, File
   }
 
   private static SegmentGeneratorConfig getSegmentGeneratorConfig(File workingDir, TableConfig tableConfig,
-      SegmentMetadataImpl segmentMetadata, String segmentName) {
-    SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, segmentMetadata.getSchema());
+      SegmentMetadataImpl segmentMetadata, String segmentName, Schema schema) {
```

Comment:
[For my learning]
Here we have two objects which can furnish the schema.
a) segmentMetadata
b) schema - passed in as argument. I understand that for the hotfix, we are adding this new variable.


Do we plan to omit the new variable in the follow up?

Reply:
> Do we plan to omit the new variable in the follow up?

Usually all minion tasks use ZK schema itself and not from segmentMetadata. So better to follow the same here as well. I wouldn't worry much about the new variable as it's a private method anyways.